cdap-security/src/main/java/co/cask/cdap/security/authorization/AuthorizerInstantiator.java

```diff
@@ -91,25 +88,33 @@
 
   @Inject
   @VisibleForTesting
-  public AuthorizerInstantiatorService(CConfiguration cConf, AuthorizationContextFactory authorizationContextFactory) {
+  public AuthorizerInstantiator(CConfiguration cConf, AuthorizationContextFactory authorizationContextFactory) {
     this.cConf = cConf;
     this.authenticationEnabled = cConf.getBoolean(Constants.Security.ENABLED);
     this.authorizationEnabled = cConf.getBoolean(Constants.Security.Authorization.ENABLED);
     this.instantiatorFactory = new InstantiatorFactory(false);
     this.authorizationContextFactory = authorizationContextFactory;
   }
 
+  /**
+   * Returns an instance of the configured {@link Authorizer} extension, or of {@link NoOpAuthorizer}, if
+   * authorization is disabled.
+   */
   @Override
-  protected void startUp() throws Exception {
+  public Authorizer get() {
+    if (authorizer != null) {
+      return authorizer;
+    }
+
     if (!authorizationEnabled) {
       LOG.debug("Authorization is disabled. Using a no-op authorizer.");
-      this.authorizer = new NoOpAuthorizer();
-      return;
+      authorizer = new NoOpAuthorizer();
```

Comment:
We remember the authorizer instance, so that next get call would return it immediately.